tests/testthat/test_generate_strings.R

```diff
@@ -0,0 +1,302 @@
+  unique_individuals <- n_groups*n_individuals
+  n_expected_calls <- n_calls*unique_individuals
+
+  # Get the number of generated calls
+  generated_calls <- nrow(generated_strings)
+
+  # Check that the correct number of calls were generated
+  expect_true(generated_calls == n_expected_calls,
+               info = "Not all generated calls have the expected number.")
+})
+
+# 3. Unit test to check that the number of groups are correct
+test_that("Generated groups are correct number", {
+
+  # Avoid library calls and other changes to the virual environment
+  # See https://r-pkgs.rg/testing-design.html
+  withr::local_package("tidyverse")
+  withr::local_package("dplyr")
+  withr::local_package("lubridate")
+
+  # Just for code development
+  # library(tidyverse)
+  # library(lubridate)
+  # library (testthat)
+
+  # Define parameters
+  string_length <- 16
+  n_calls <- 10
+  n_groups <- 2
+  n_individuals <- 5
+
+  # Call the function with the test parameters
+  generated_strings <- generate_strings(n_groups = n_groups, n_individuals = n_individuals, n_calls = n_calls, string_length = string_length, group_information = 8, individual_information = 2)
+
+  # glimpse(generated_strings)
+
+  # Extract the number of unique groups
+  unique_groups <- length(unique(generated_strings$Group))
+
+  # Check that the number of groups are correct
+  expect_true(all(unique_groups == n_groups),
+              info = "Not all number of groups were correctly generated.")
+})
+
+# 4. Unit test to check that the number of individuals are correct
+test_that("Generated number of individuals are correct", {
+
+  # Avoid library calls and other changes to the  virtual environment
+  # See https://r-pkgs.org/testing-design.html
+  withr::local_package("tidyverse")
+  withr::local_package("dplyr")
+  withr::local_package("lubridate")
+
+  # Just for code development
+  # library(tidyverse)
+  # library(lubridate)
+  # library(testthat)
+
+  # Define parameters
+  string_length <- 16
+  n_calls <- 10
+  n_groups <- 2
+  n_individuals <- 5
+
+  # Call the function with the test parameters
+  generated_strings <- generate_strings(n_groups = n_groups, n_individuals = n_individuals, n_calls = n_calls, string_length = string_length, group_information = 8, individual_information = 2)
+
+  # glimpse(generated_strings)
+
+  # Get the number of unique individuals
+  unique_individuals <- length(unique(generated_strings$Individual))
+
+  # Check that the correct the number of individuals are correct
+  expect_equal(unique_individuals, n_individuals,
+               info = "Not all numbers of individuals were generated correctly.")
+})
+
+# 5. Unit test to check that the number of characters in each string devoted to group information is correct
+test_that("Generated number of characters in each string devoted to group information have the correct length", {
+
+  # Avoid library calls and other changes to the virtual environment
```

Comment:
Let's remove unit tests 5 and 6 here. I'm pretty sure that these are the unit tests that Raneem wrote out in another pull request